### PR TITLE
[homekit] fix for battery charging state

### PR DESCRIPTION
**Pull request by yfre**
the battery HomeKit accessory type was not always working correctly as it expect charging state characteristic depending on battery type. Only chargeable battery have this characteristic and it is mandatory for them.

It is first case of conditionally mandatory characteristic so that some code changes were needed. 

This PR 
- add supports for conditional mandatory characteristic. currently only one - battery charging state
- improve logging and some function names
- update documentation with an example for sensor with battery
- resize large images

Signed-off-by: Eugen Freiter <freiter@gmx.de>

### CODE REVIEW

**Review by lolodomo — APPROVED** [review]
LGTM

File: bundles/org.openhab.io.homekit/README.md [hunk 1]
```diff
@@ -517,12 +518,17 @@ Group           FanWithLight        "Fan with Light"                           {
 
 ![ui_fan_with_light_primary.png](doc/ui_fan_with_light_primary.png)
 
+Similarly, you can create a sensor with battery
+
+![ui_sensor_with_battery.png](doc/ui_sensor_with_battery.png)
```
Comment (lolodomo) [inline bundles/org.openhab.io.homekit/README.md:523]: This image is missing?

Comment (yfre) [reply bundles/org.openhab.io.homekit/README.md:523]: good one. thank you. added.

**Review by lolodomo — APPROVED** [review]
LGTM

**Pull request merged**